### ISSUE TITLE
`pdf`, `cdf`, etc. do not work with `OffsetArrays`

### ISSUE THREAD

**Issue by adityam** [JuliaStats/Distributions.jl#653]
For univariate distributions, the functions `pdf`, `cdf`, etc. do not work with offset arrays. Here is an example illustrating this:

~~~
using OffsetArrays
using Distributions

const N = 10 
const L = 5 

const range = linspace(-L, L, 2N+1)

const A = collect(range)
const B = OffsetArray(range, -N:N)

W = Normal(0,1)

X = cdf(W, A)
Y = cdf(W, B)
~~~

I get the following error:

~~~
ERROR: LoadError: size not supported for arrays with indices (-10:10,); see http://docs.julialang.org/en/latest/devdocs/offset-arrays/
~~~

The linked page says that the use of `size` should be replaced by `indices` for the code to work with general arrays.

**Comment by andreasnoack** [comment]
That is true. That package has never been updated to support that. It would be great if you could open a PR that fixes this.

**Comment by adityam** [comment]
I created a pull request. I do not completely understand the macro programming code, so I am hoping that the test cases will catch any serious errors.

**Comment by matbesancon** [comment]
This has been fixed a long time ago by #655
